botocore/parsers.py

```diff
@@ -694,10 +695,24 @@ def _do_error_parse(self, response, shape):
         error['Error']['Message'] = body.get(
             'message', body.get('Message', '')
         )
+        # Error response may contain an x-amzn-query-error header for json
+        # we need to fetch the error code from this header in that case
+        query_error = header.get('x-amzn-query-error') if header else None
+        query_error_code = None
+        if (
+            query_error
+            and query_error.split(';')
+            and len(query_error.split(';')) == 2
+        ):
+            query_error_code = query_error.split(';')[0]
```

Comment:
Let's avoid splitting this multiple times, we can simplify to just:

```suggestion
        query_error_components = query_error.split(';')
        code = None
        if len(query_error_components) == 2:
            code = query_error_components[0]
        if code is None:
            code =  body.get('__type', response_code and str(response_code))   
```

Reply:
Thanks. Changes made accordingly.